ckanext/datastorer/tasks.py

```
@@ -165,6 +156,20 @@ def send_request(data):
                          )
         check_response_and_retry(response, datastore_create_request_url, logger)
 
+    # Delete any existing data before proceeding. Otherwise 'datastore_create' will
+    # append to the existing datastore. And if the fields have significantly changed,
+    # it may also fail.
+    try:
+        logger.info('Deleting existing datastore (it may not exist): {0}.'.format(resource['id']))
+        response = requests.post('%s/api/action/datastore_delete' % (ckan_url),
+                        data=json.dumps({'resource_id': resource['id'],}),
+                         headers={'Content-Type': 'application/json',
+                                  'Authorization': context['apikey']},
+                         )
+        check_response_and_retry(response, datastore_create_request_url, logger)
+    except:
```

This is potentially dangerous and I'd like to have either a real check before we delete anything or a check for the exact exception that is expected. Even better, if there is an exception that matches but the error code is not the expected 404, we should raise the error. 

See http://docs.python-requests.org/en/latest/user/quickstart/#errors-and-exceptions for more details on requests errors.